servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -2213,7 +2213,7 @@ public final Publisher<T> concatPropagateCancel(Completable next) {
      * @see #retry(boolean, BiIntPredicate)
      */
     public final Publisher<T> retry(BiIntPredicate<Throwable> shouldRetry) {
-        return retry(false, shouldRetry);
+        return retry(true, shouldRetry);
```

Comment:
While I'm +1 for this default behavior change, it's disappointing that reasons to have an extra overload that takes a boolean are:
1. Other RS implementations behave like terminateOnNextException == false
2. It's a behavior change for existing users.

Reply:
agreed with 1 & 2 but also there are legit cases (you are subscribing, and your subscriber throws due to existing application logic) that would be difficult to orchestrate control flow w/out this option.